src/pages/promo-code/tests/promo-code-page.spec.js

```diff
@@ -1,47 +1,136 @@
+import { MockedProvider } from '@apollo/client/testing';
 import PromoCodePage from '../promo-code-page';
 import { theme } from '../../../components/app/app-theme/app.theme';
-import { promoCodes } from './promo-code-page.variables';
 import TableContainerRow from '../../../containers/table-container-row';
 import TableContainerGenerator from '../../../containers/table-container-generator';
 import LoadingBar from '../../../components/loading-bar';
+import { getAllPromoCodes } from '../operations/promo-code.queries';
+import { deletePromoCodeByID } from '../operations/promo-code.mutation';
 
 jest.mock('react-redux');
-jest.mock('@apollo/client');
 jest.mock('../promo-code-page.styles', () => ({
   useStyles: () => ({})
 }));
 jest.mock('connected-react-router', () => ({
   push: jest.fn()
 }));
+const rowsPerPage = 10;
+const currentPage = 1;
+
+const mockStore = {
+  rowsPerPage: 10,
+  currentPage: 1
+};
+
+useSelector.mockImplementation(() => mockStore);
+const mocks = [
+  {
+    request: {
+      query: getAllPromoCodes,
+      variables: {
+        limit: rowsPerPage,
+        skip: rowsPerPage * currentPage
+      }
+    },
+    result: {
+      data: {
+        getAllPromoCodes: {
+          count: '1',
+          items: [
+            {
+              categories: ['backpacks', 'bags', 'accessories', 'constructor'],
+              code: 'www',
+              dateFrom: '2022-04-08T14:22:43.437Z',
+              dateTo: '2022-04-30T14:22:43.452Z',
+              discount: 10,
+              _id: '62470ac47cc7af17305abd47'
+            }
+          ]
+        }
+      },
+      loading: false
+    }
+  },
+  {
+    request: {
+      query: deletePromoCodeByID,
+      variables: {
+        id: 'promoID'
+      },
+      context: {
+        headers: {
+          token:
+            'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiI2MjJmNDhjZWEwZDEyYTAwMjRkNzgyN2EiLCJpYXQiOjE2NDk0MjkyNzAsImV4cCI6MTY0OTQzMjg3MH0.WcDQSLKvEpOCIKe_4D6DrCgNhUd0c4Hgyyo75hfEhnI'
+        }
+      }
+    },
+    result: {
+      data: {
+        deletePromoCode: { _id: '6239a310ed27613a04a17bb8' }
+      }
+    }
+  }
+];
+const mocks2 = [
+  {
+    request: {
+      query: getAllPromoCodes,
+      variables: {
+        limit: rowsPerPage,
+        skip: rowsPerPage * currentPage
+      }
+    },
+    result: {
+      data: {
+        getAllPromoCodes: {
+          count: '0',
+          items: []
+        }
+      },
+      loading: false
+    }
+  },
+  {
+    request: {
+      query: deletePromoCodeByID,
+      variables: {
+        id: 'promoID'
+      },
+      context: {
+        headers: {
+          token:
+            'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiI2MjJmNDhjZWEwZDEyYTAwMjRkNzgyN2EiLCJpYXQiOjE2NDk0MjkyNzAsImV4cCI6MTY0OTQzMjg3MH0.WcDQSLKvEpOCIKe_4D6DrCgNhUd0c4Hgyyo75hfEhnI'
+        }
+      }
+    },
+    result: {
+      data: {
+        deletePromoCode: { _id: '6239a310ed27613a04a17bb8' }
+      }
+    }
+  }
+];
```

Comment:
It will probably be better if you take mocks to another file (promo-code-page.variables.js)